src/scss/_mixins.scss

```diff
@@ -1,80 +1,37 @@
-////
-/// @group o-table
-/// @link http://registry.origami.ft.com/components/o-table
-////
+/// Outputs all features and styles available.
+@mixin oTableAll() {
+	// Basic styles and simple utilities.
+	@include oTableBase;
 
-/// Table styles boilerplate
-/// Generates all styles available, with a set of modifiers for alternative styles
-///
-/// @param {string} $classname
-@mixin oTableAll($classname: o-table) {
-	.#{$classname}-wrapper {
-		@include oTableWrapper;
-	}
-	.#{$classname} {
-		@include oTableBase;
-		.#{$classname}__caption,
-		// Todo: remove these
-		.#{$classname}__caption--top,
-		.#{$classname}__caption--bottom {
-			@include oTableCaption;
-		}
-		.#{$classname}__caption--top {
-			@include oTableCaptionTop;
-		}
-		.#{$classname}__caption--bottom {
-			@include oTableCaptionBottom;
-		}
-
-		.#{$classname}__cell--numeric {
-			@include oTableCellNumeric;
-		}
+	// Sortable columns.
+	@include oTableSort;
 
-		.#{$classname}__cell--content-secondary {
-			@include oTableCellContentSecondary;
-		}
-
-		.#{$classname}__cell--vertically-center {
-			@include oTableCellVerticallyCenter;
-		}
-	}
-	.#{$classname}--horizontal-lines {
-		@include oTableHorizontalLines;
-	}
-	.#{$classname}--vertical-lines {
-		@include oTableVerticalLines;
-	}
-	.#{$classname}--horizontal-borders {
-		@include oTableHorizontalBorders;
-	}
-	.#{$classname}--vertical-borders {
-		@include oTableVerticalBorders;
-	}
-	.#{$classname}--responsive-flat {
-		@include oTableResponsiveFlat;
-	}
-	.#{$classname}--responsive-overflow {
-		@include oTableResponsiveOverflow;
-	}
-	.#{$classname}--responsive-scroll {
-		@include oTableResponsiveScroll;
-	}
+	// Respnsive solutions.
+	@include oTableWrapper;
+	@include oTableContainer;
+	@include oTableControlsOverlay;
+	@include oTableFadeOverlay;
+	@include oTableResponsiveFlat;
+	@include oTableResponsiveScroll;
 
-	@if _oTableSupports('row-headings') {
-		.#{$classname}--row-headings {
-			@include oTableRowHeadings;
-		}
-	}
+	// Lines.
+	@include oTableHorizontalLines;
+	@include oTableVerticalLines;
+	@include oTableHorizontalBorders;
+	@include oTableVerticalBorders;
 
+	// Compact.
 	@if _oTableSupports('compact') {
-		.#{$classname}--compact {
-			@include oTableCompact;
-		}
+		@include oTableCompact;
 	}
 
+	// Stripes.
 	@if _oTableSupports('stripe') {
```

Comment:
#überpedantic but stripe could be stripes if the mixin is stripes and a row has many stripes? 😬